### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by AdamOndrejkovic** [microsoft/monaco-editor#3697] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [X] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Actual Behavior

It takes up to 5 seconds to load which slows the site

### Expected Behavior

Is it possible to get it to 2 seconds or less

### Additional Context

Hi, I am using Monaco editor for a project however it takes quite a while to load. In my code, I even define the language which when I looked should improve the loading time however it still hasn't improved. Is there anything else that is possible to do so that the performance improves? Maybe some settings in JSON or in the editor itself. I couldn't find any relevant reference to it.

**Comment by hediet** [comment]
Can you try to figure out if there is anything that does not need to be loaded in your case?

We always try to have a fast loading editor, but your feature request is quite unspecific.

**Reply by pantonis** [reply]
We have issues with loading the editor initially. It takes 5 seconds to show up on the first time. We found out that editor.main.js which is about 3.5MB takes 4 seconds to load. as soon as this loads editor becomes available

**Comment by AdamOndrejkovic** [comment]
Then maybe just my configuration is wrong. How can I test if everything I am using is required and if not then disable it to use just the bare minimum for it to make it work and perform better?

**Comment by hediet** [comment]
You can use the dev tools of your browser to see which scripts get requested.
Or check webpack which sources are bundled.

**Comment by AdamOndrejkovic** [comment]
Monaco-editor/esm/vs gets loaded in the webpack other than that it's a bunch of different scripts
